nmigen/hdl/dsl.py

```diff
@@ -418,7 +425,18 @@ def _add_submodule(self, submodule, name=None):
         if not hasattr(submodule, "elaborate"):
             raise TypeError("Trying to add '{!r}', which does not implement .elaborate(), as "
                             "a submodule".format(submodule))
-        self._submodules.append((submodule, name))
+        if name == None:
+            self._anon_submodules.append(submodule)
+        else:
+            if name in self._named_submodules:
+                raise ValueError("Submodule named '{}' already exists".format(name))
```

Comment:
Actually, I wasn't sure if this should be a ValueError or an AttributeError or maybe a SyntaxError?

Reply:
NameError?